tools/shell/embedded_shell.cpp

```diff
@@ -176,6 +186,27 @@ void findTableVariableNames(const std::string buf, std::regex tableRegex,
     }
 }
 
+void keywordCompletion(std::string buf, std::string prefix, std::string keyword,
+    linenoiseCompletions* lc) {
+    std::string bufEscaped = regex_replace(buf, specialChars, R"(\$&)");
+    if (regex_search(keyword, std::regex("^" + bufEscaped, std::regex_constants::icase))) {
+        std::string transformedKeyword = keyword;
+        for (size_t i = 0; i < buf.size() && i < keyword.size(); ++i) {
+            if (islower(buf[i])) {
+                transformedKeyword[i] = tolower(keyword[i]);
```

Comment:
I don't know if it makes sense to keep a lower version and upper version of keyword list make sense or not?

Reply:
This part of the code makes sure that the keyword we are completing matches the case typed by the user when we do the autocomplete. Without this, if "Ma" was typed, since our keyword is "MATCH", later on in the linenoise code this autocompletion would be rejected

Reply:
I see. Sounds good.